toolchain/parse/typed_nodes.h

```diff
@@ -72,6 +72,19 @@ struct LeafNode {
 // - Any provided `Aggregate` type that is a simple aggregate type such as
 // `struct Aggregate { T x; U y; }`,
 //   to match children with types `T` and `U`.
+//
+// Each parse node should also have exactly one field of type `Lex::*TokenIndex`
+// that matches the token corresponding to the parse node itself. This is not a
+// child node in the parse tree, but specifies the token kind associated with
+// the parse node. The location of the field relative to the child nodes
+// indicates the location within the corresponding grammar production where the
+// token appears.
+//
+// There should be exactly one `Lex::*TokenIndex` field in each parse node, and
+// it should always have the name `token`. In the case where a parse node can
+// correspond to any kind of token, `Lex::TokenIndex` can be used instead. This
+// should only be used when the node kind is either not used in a finished tree,
+// such as `Placeholder`, or is always invalid, such as `InvalidParse`.
```

Comment:
This looks duplicative: both paragraphs start with a statement that each needs to have one `TokenIndex` field. The latter paragraph specifies it needs to be `token`, the former specifies it's not part of the parse tree. Should the paragraphs be unified, or one deleted?

Reply:
Yeah, there's some redundancy here. Rephrased.